### PR TITLE
fix: Node.js 10 support

### PR DESCRIPTION
**Pull request by jacobheun**
Opening this early to provide more visibility.

**Update**:
The jenkins builds are now running and working for node 10.0. There is an issue in node 10.1 with fs.promises warnings, mentioned below, but there is a fix already merged that should go out with the next release so I am going to leave that for now.

I will be working on getting all the dependency change PRs merged so I can get this PR finalized.

**Things to fix**:
- [x] Get npm install working on node 10
- [x] Resolve [repo not initialized errors](https://github.com/ipfs/js-ipfs/pull/1338)
- [x] Resolve fs.promises ExperimentalWarning in node 10.1 (these go to stderr and cause any exec spawn tests to fail) This is fine in node 10.0 and should be fixed in the next version of node 10 (https://github.com/nodejs/node/pull/20632)
- [x] Get yarn install working for CI (incorrect dependencies being installed)
- [x] Resolve failing tests
- [x] (PR Pending) Fix windows, [node 10 failure in lock-me](https://github.com/dignifiedquire/lock-me/pull/3), due to node 10 fs module bug.

## Dependencies
The following PRs need to be completed prior to this release.
- [x] ipfs/js-datastore-fs#16 merged and released
- [x] ipfs/js-datastore-level#7 merged and released
  - [x] ipfs/js-ipfs-repo#168 merged and released, depends on fs and level PRs    
    - [x] ipld/js-ipld#129 merged and released, causes an issue with npm install of old leveldown

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -110,10 +110,10 @@
     "ipfs-block": "~0.7.1",
     "ipfs-block-service": "~0.14.0",
     "ipfs-multipart": "~0.1.0",
-    "ipfs-repo": "~0.20.0",
+    "ipfs-repo": "^0.22.1",
```
Comment (fsdiogo) [inline package.json]: We should keep the `~` for packages below `1.0.0`, as stated in the [js-ipfs guidelines](https://github.com/ipfs/community/blob/master/js-code-guidelines.md#dependency-versions).

Comment (diasdavid) [reply package.json]: good catch! :)

**Comment by alanshaw** [comment]
👀 ing now

**Comment by alanshaw** [comment]
@jacobheun could you setup commit signing for next time? https://help.github.com/articles/signing-commits-using-gpg/

**Pull request merged**